xmppserver/src/main/java/org/jivesoftware/openfire/OfflineMessageStore.java

```diff
@@ -563,4 +619,154 @@ static boolean shouldStoreMessage(final Message message) {
+                    {
+                        if (ClusterManager.isSeniorClusterMember())
+                        {
+                            if (deleteOldOfflineMessagesFromDB())
+                            {
+                                readSizeForAllUsers();
+                            }
+                        }
+                    }
+                    else
+                    {
+                        if (deleteOldOfflineMessagesFromDB())
+                        {
+                            readSizeForAllUsers();
+                        }
+                    }
+                } catch (Exception e) {
+                    Log.error("Offline message cleaning - Could not set timer for check interval!", e.fillInStackTrace());
+                }
+            }
+        }, 10 * 1000, checkinterval * 60 * 1000); //starts after 10 seconds and repeat...
+    }
+
+    private void cancelTimer() {
+        if (timer != null) {
+            Log.info("Offline message cleaning - Stop old timer if started");
+            try {
+                timer.cancel();
+                timer.purge();
+            } catch (Exception e) {
+                Log.warn("Offline message cleaning - Could not stop the timer!", e.fillInStackTrace());
+            }
+        }
+        timer = null;
+    }
+
+    public void readSizeForAllUsers() {
+        // See if the size is cached.
+        sizeCache.clear();
+
+        int size = 0;
+        String username = null;
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = DbConnectionManager.getConnection();
+            pstmt = con.prepareStatement(SELECT_SIZE_OFFLINE_ALL_USERS);
+
+            rs = pstmt.executeQuery();
+            while (rs.next()) {
+                size = rs.getInt(1);
+                username = rs.getString(2);
+                // Add the value to cache.
+                sizeCache.put(username, size);
+            }
+        }
+        catch (Exception e) {
+            Log.error(LocaleUtils.getLocalizedString("admin.error"), e);
+        }
+        finally {
+            DbConnectionManager.closeConnection(rs, pstmt, con);
+        }
+    }
+
+    private boolean deleteOldOfflineMessagesFromDB() {
+
+        Log.info("Offline message cleaning - Deleting offline messages older than "+String.valueOf(daystolive)+" days.");
+
+        Connection con = null;
+
+        PreparedStatement pstmt = null;
+        try {
+            con = DbConnectionManager.getConnection();
+
+            pstmt = con.prepareStatement(DELETE_OFFLINE_MESSAGE_BEFORE);
+
+            long now = System.currentTimeMillis();
+            long delta = daystolive*24*60*60*1000;
+            long pastTime = now - delta;
+            String creationDatePast = String.format ("%015d", pastTime);
+
+            pstmt.setString(1,creationDatePast);
+
+            if (pstmt.execute())
```

Comment:
Please use `{` and `}` even for single-line blocks. Errors are easy to make without them.